### PR TITLE
Feature/add help command

### PR DESCRIPTION
**Pull request by wildman777**
For l'issue #12 

-  Use the champ active to check if the function is activated or disabled
- Need to add the champ description for the description of the function listed
- Need to be tested

### CODE REVIEW

File: src/functions/help/main.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from dbm import _Database
```
Comment (ArmandBriere) [inline src/functions/help/main.py:1]: Delete useless import

File: src/functions/help/main.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import functions_framework
+from google.cloud.firestore_v1.collection import CollectionReference
+from google.cloud.firestore_v1.document import DocumentReference
+from google.cloud.firestore_v1.client import Client
+from google.cloud.firestore_v1.base_document import DocumentSnapshot
+
+@functions_framework.http
+def hello(request) -> Tuple[str, int]:
+    """Return the requested list of all actived function"""
```
Comment (ArmandBriere) [inline src/functions/help/main.py:12]: missing `.` at the end

File: src/functions/help/main.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    request_json: Optional[Any] = request.get_json(silent=True)
+
+    if request_json:
+        server_id = request_json.get("server_id", None)
+        
+        database: Client = Client(project="archy-f06ed")
+        
+        function_collection: CollectionReference = database.collection("servers").document(server_id).collection("function")
```
Comment (ArmandBriere) [inline src/functions/help/main.py:21]: add `s` to `function`

File: src/functions/help/main.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        doc_ref = function_collection.collection("functions")
```
Comment (ArmandBriere) [inline src/functions/help/main.py:22]: delete

File: src/functions/help/main.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        docs: DocumentSnapshot = doc_ref.where(u'active',u'==', True ).stream()
```
Comment (ArmandBriere) [inline src/functions/help/main.py:23]: run `black -l 120 .` in `src` folder

File: src/functions/help/main.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        
+        if not server_id:
+            print("Exit: Missing data")
+            return "", 200
+        
+        if docs.exists:
+            return list(docs), 200
```
Comment (ArmandBriere) [inline src/functions/help/main.py:30]: `return list(docs), 200` -> `return [doc.get("description") for doc in docs]`
but we want a string of description and not just a list so we need to str concat the return

**Review by ArmandBriere — CHANGES_REQUESTED** [review]

File: src/functions/help/main.py (hunk 4 above)
Comment (wildman777) [reply src/functions/help/main.py:22]: delete ?

File: src/functions/help/main.py (hunk 2 above)
Comment (wildman777) [reply src/functions/help/main.py]: done

File: src/functions/help/main.py (hunk 3 above)
Comment (wildman777) [reply src/functions/help/main.py]: done

File: src/functions/help/main.py (hunk 6 above)
Comment (wildman777) [reply src/functions/help/main.py]: done

**Review by wildman777 — COMMENTED** [review]
Fix done:

- Deleted useless import.
- Add . to """Return the requested list of all actived function."""
- Change return list(docs) to return [doc.get("description") for doc in docs].
- Add s to functions in collection.
- Didn't know what you wanted to delete when you say delete.

File: src/functions/help/main.py (hunk 1 above)
Comment (wildman777) [reply src/functions/help/main.py]: done

**Review by wildman777 — COMMENTED** [review]
Fix:
- return a concat string in newline --> '\n'.join([doc.get("description") for doc in docs])

File: src/functions/help/main.py (hunk 5 above)
Comment (wildman777) [reply src/functions/help/main.py]: I have now formatted with black

File: src/functions/help/main.py (hunk 6 above)
Comment (ArmandBriere) [reply src/functions/help/main.py]: well yeah but what do we want this command to return ?
We want something nice for the user to understand how to use the commands

ex:
```bash
!help
>
!hello -> Greet the user back
!froge -> Send a random froge emoji selected from the server
!level -> Return the user's level
!level @username -> Return the selected user's level
... 
```

Something like this for a starter would be nice, we could improve the layout and design later

right now we are returning:

```bash
[ "Greet the user back","Send a random froge emoji selected from the server"...] 
```

File: src/functions/help/main.py (hunk 4 above)
Comment (wildman777) [reply src/functions/help/main.py:23]: Alright! I will remove and bring the change asked, "964701887540645908" should be server_id.

File: src/functions/help/main.py (hunk 6 above)
Comment (wildman777) [reply src/functions/help/main.py]: Alright! the change will should match the example 
id -> function name
description -> description of the function

**Review by wildman777 — COMMENTED** [review]
Fix:
- Formatted with black
- return is a concat f-string  with newline each function with his name -> his description ex: !level -> Return the user's level
- remove doc_ref for Generator as asked

**Review by wildman777 — COMMENTED** [review]
Fix:
- Import are now correctly sorted/formatted by isort
- pass all the test

**Review by ArmandBriere — APPROVED** [review]

**Pull request merged**